java/vector/src/main/java/org/apache/arrow/vector/IntervalDayVector.java

```diff
@@ -126,15 +127,14 @@ public void get(int index, NullableIntervalDayHolder holder) {
    * @param index   position of element
    * @return element at given index
    */
-  public Period getObject(int index) {
+  public Duration getObject(int index) {
     if (isSet(index) == 0) {
       return null;
     } else {
       final int startIndex = index * TYPE_WIDTH;
       final int days = valueBuffer.getInt(startIndex);
       final int milliseconds = valueBuffer.getInt(startIndex + MILLISECOND_OFFSET);
-      final Period p = new Period();
-      return p.plusDays(days).plusMillis(milliseconds);
+      return Duration.ofDays(days).plusMillis(milliseconds);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API.